zips/zip-0048.md

```diff
@@ -1,9 +1,195 @@
+  - This is the BIP that essentially all hardware wallets use for P2WSH and P2SH-P2WSH on
+    any Bitcoin-compatible chain.
+
+- BIP 87 [^bip-0087] uses `m / purpose' / coin_type' / account' / change / address_index`
+  - The problem is that this BIP does not consider key reuse across different scripts
+    (which are to some degree different payment protocols) to be an issue, whereas we do.
+  - This BIP is essentially unused by hardware wallets (which instead implement BIPs 45
+    and 48).
+
+While there is verifiable usage of P2SH multisig on the Zcash chain, there has been no
+standard approach for public deployments (likely due to the above issues), and thus no
+support for it in Zcash-compatible hardware wallets. This ZIP fills the gap.
+
+
+# Privacy Implications
+
+TBD
+
+
+# Requirements
+
+TBD
+
+
+# Specification
+
+This ZIP applies to both Mainnet and Testnet, using the same $\mathit{coin\_type}$
+constants as defined in [^slip-0044] where applicable. Note that in keeping with that
+document, all cryptocurrency testnets share $\mathit{coin\_type}$ index 1.
+
+## Output descriptor and script
+
+TODO: Specify how P2MS scripts are constructed for Zcash addresses and transactions,
+referencing the following BIPs:
+
+- BIP 379 [^bip-0379]: Miniscript
+- BIP 380 [^bip-0380]: Output Script Descriptors General Operation
+- BIP 381 [^bip-0381]: Non-Segwit Output Script Descriptors
+- BIP 383 [^bip-0383]: Multisig Output Script Descriptors
+  - Implies BIP 67 [^bip-0067]
+- BIP 388 [^bip-0388]: Wallet Policies for Descriptor Wallets
+
+Below here is proposed specification text:
+
+Zcash wallets SHOULD use the BIP 383 [^bip-0383] `sortedmulti()` script expression to
+produce Zcash P2MS scripts, which would then be contained in the BIP 381 [^bip-0381]
+`sh()` script expression to produce Zcash P2SH addresses.
+
+The following BIP 388 [^bip-0388] descriptor template produces a 2-of-3 P2SH multisig
+wallet compatible with this ZIP:
+
+```
+sh(sortedmulti(2,@0/**,@1/**,@2/**))
+```
+
+## Hierarchical Derivation
+
+Given a set of participants that each have some seed material (e.g. BIP 39 [^bip-0039]
+mnemonic phrases), and a BIP 380 [^bip-0380] output descriptor that encodes the desired
+multisig policy, wallets need to produce a sequence of P2SH addresses that can be given
+out for receiving funds, and scanned to detect incoming payments. This specification
+adopts BIP 48 [^bip-0048] for this purpose, with the following modification:
+
+- The only defined $\mathit{script\_type}$ is $\mathtt{0x133000}$ for “Zcash P2SH” as
+  defined in this specification. The examples for Nested Segwit and Native Segwit are
+  therefore not applicable.
+
+## Transaction construction
+
+TODO: Anything we need to specify here?
+
+
+# Rationale
+
+We specify a likely-non-colliding $\mathit{script\_type}$ instead of using either `0` (the gap
+left in BIP 48) or `2` (the next unassigned contant in BIP 48), because both of those values
```

Comment:
```suggestion
We specify a likely-non-colliding $\mathit{script\_type}$ instead of using either $0'$ (the gap
left in BIP 48) or $2'$ (the next unassigned contant in BIP 48), because both of those values
```